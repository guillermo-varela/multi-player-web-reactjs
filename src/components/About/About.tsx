import React from 'react';
import Jumbotron from 'react-bootstrap/Jumbotron';
import { FormattedMessage } from 'react-intl';

const About: React.FC = () => {
  return (
    <Jumbotron>
      <h1>
        <FormattedMessage id='component.about.hello' />
      </h1>
      <br />
      <p>
        <FormattedMessage id='component.about.introduction' />
      </p>
      <ul>
        <li>
          <a href="https://reactjs.org/" target="_blank" rel="noopener noreferrer">
            React
          </a>
        </li>
        <li>
          <a href="https://reactjs.org/docs/hooks-intro.html" target="_blank" rel="noopener noreferrer">
            Hooks
          </a>
        </li>
        <li>
          <a href="https://reacttraining.com/react-router/web/" target="_blank" rel="noopener noreferrer">
            React Router
          </a>
        </li>
        <li>
          <a href="https://react-bootstrap.github.io/" target="_blank" rel="noopener noreferrer">
            React Bootstrap
          </a>
        </li>
        <li>
          <a href="https://github.com/react-bootstrap/react-router-bootstrap" target="_blank"
            rel="noopener noreferrer">
            react-router-bootstrap
          </a>
        </li>
        <li>
          <a href="https://sass-lang.com/" target="_blank"
            rel="noopener noreferrer">
            SASS
          </a>
        </li>
        <li>
          <a href="https://reactjs.org/docs/code-splitting.html#route-based-code-splitting" target="_blank"
            rel="noopener noreferrer">
            <FormattedMessage id='component.about.codeSplit' />
          </a>
        </li>
        <li>
          <a href="https://redux.js.org/recipes/usage-with-typescript/" target="_blank" rel="noopener noreferrer">
            <FormattedMessage id='component.about.redux.typescript' />
          </a>
        </li>
        <li>
          <a href="https://jaredpalmer.com/formik" target="_blank" rel="noopener noreferrer">
            <FormattedMessage id='component.about.forms.formik' />
          </a>
        </li>
        <li>
          <a href="https://github.com/formatjs/react-intl" target="_blank" rel="noopener noreferrer">
            <FormattedMessage id='component.about.internationalization' />
          </a>
        </li>
        <li>
          <a href="https://docs.spring.io/spring/docs/current/spring-framework-reference/web-reactive.html"
            target="_blank" rel="noopener noreferrer">
            <FormattedMessage id='component.about.rest.webflux' />
          </a>
        </li>
        <li>
          <a href="https://connect2id.com/products/nimbus-jose-jwt" target="_blank" rel="noopener noreferrer">
            <FormattedMessage id='component.about.jwt' />
          </a>
        </li>
        <li>
          <a href="https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#use-of-custom-request-headers"
            target="_blank" rel="noopener noreferrer">
            <FormattedMessage id='component.about.csrf' />
          </a>
        </li>
      </ul>
      <p>
        <FormattedMessage id='component.about.code' />
      </p>
      <ul>
        <li>
          Backend:&nbsp;
            <a href="https://github.com/guillermo-varela/multi-player-auth" target="_blank" rel="noopener noreferrer">
              https://github.com/guillermo-varela/multi-player-auth
            </a>
        </li>
        <li>
          Frontend:&nbsp;
            <a href="https://github.com/guillermo-varela/multi-player-web-reactjs" target="_blank" rel="noopener noreferrer">
              https://github.com/guillermo-varela/multi-player-web-reactjs
            </a>
        </li>
      </ul>
      <p>
        <FormattedMessage id='component.about.pixabay'
          values={{
            link:
              <a href="https://pixabay.com/api/docs" target="_blank" rel="noopener noreferrer">
                <FormattedMessage id='component.about.pixabay.name' />
              </a>
          }} />
      </p>
    </Jumbotron>
  );
}

export default About;
