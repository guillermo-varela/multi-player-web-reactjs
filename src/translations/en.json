{
  "component.about.code": "The source code for this project can be found at:",
  "component.about.codeSplit": "Route based code splitting",
  "component.about.csrf": "Use of custom HTTP headers in stateless/sessionless environments to prevent CSRF attacks",
  "component.about.forms.formik": "Forms in React using Formik",
  "component.about.hello": "Hi again :)",
  "component.about.internationalization": "Support for multiple languages (internationalization) using react-intl",
  "component.about.introduction": "This web application was developed to learn about:",
  "component.about.jwt": "Stateless/sessionless authentication on a server using \"Javascript Object Signing and Encryption\" (JOSE) and \"JSON Web Token\" (JWT)\"",
  "component.about.openAPI": "OpenAPI 3",
  "component.about.pixabay": "The multimedia content was fetched using the {link}.",
  "component.about.pixabay.name": "Pixabay API",
  "component.about.redux.typescript": "Redux using TypeScript",
  "component.about.rest.webflux": "REST services with reactive architecture (non-blocking) with Spring WebFlux and Spring Security",
  "component.error.message": "Ops! An error occurred...",
  "component.signin.button.sending": "Sending...",
  "component.signin.button.submit": "Sign In",
  "component.signin.credentials.hint": "Use user@test.com as email and password to test",
  "component.signin.empty.email": "Enter an email",
  "component.signin.incorrect.credentials": "Incorrect credentials",
  "component.signin.invalid.email": "Enter a valid email",
  "component.signin.invalid.password": "Enter a password",
  "component.signin.label.email": "Email",
  "component.signin.label.password": "Password",
  "menu.about": "About",
  "menu.home": "Home",
  "menu.images": "Images",
  "menu.more": "View More",
  "menu.sections": "Sections",
  "menu.signIn": "Sign In",
  "menu.signOut": "Sign Out",
  "menu.signOut.error.message": "Sign out failed",
  "menu.signOut.error.title": "Ops!",
  "menu.videos": "Videos",
  "multimedia.source": "Content from ",
  "multimedia.type.image": "Image",
  "multimedia.type.video": "Video",
  "nature.hello": "Hello!",
  "nature.introduction": "On this site you can find multimedia content about nature.",
  "nature.multimedia.endOfContent": "That's all for now...",
  "nature.title": "Nature",
  "root.loading": "Loading...",
  "root.locale.label": "Language",
  "root.nature.description": "Images and videos of flora and fauna.",
  "root.space.description": "Images and videos of the universe.",
  "space.hello": "Hello!",
  "space.images.introduction": "Multiple images related to outer space.",
  "space.introduction": "On this site you will find multimedia content about outer space.",
  "space.menu.title": "Space",
  "space.title": "Outer Space",
  "space.videos.introduction": "Mind-blowing videos from outer space."
}
