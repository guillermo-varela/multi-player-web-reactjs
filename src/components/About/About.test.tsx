import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Jumbotron from 'react-bootstrap/Jumbotron';
import { IntlProvider } from 'react-intl';
import en from '../../translations/en.json';
import About from './About';

describe('<About />', () => {
  it('renders <Jumbotron />', () => {
    const wrapper = mount(buildComponent());
    expect(wrapper.find(Jumbotron)).toHaveLength(1);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(buildComponent()).toJSON();
    expect(tree).toMatchSnapshot();
  });

  const buildComponent = () => {
    return (
      <IntlProvider locale='en' messages={en}>
        <About />
      </IntlProvider>
    );
  };
});
