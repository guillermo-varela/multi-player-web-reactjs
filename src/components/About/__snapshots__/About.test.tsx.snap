// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> matches snapshot 1`] = `
<div
  className="jumbotron"
>
  <h1>
    Hi again :)
  </h1>
  <br />
  <p>
    This web application was developed to learn about:
  </p>
  <ul>
    <li>
      <a
        href="https://reactjs.org/"
        rel="noopener noreferrer"
        target="_blank"
      >
        React
      </a>
    </li>
    <li>
      <a
        href="https://reactjs.org/docs/hooks-intro.html"
        rel="noopener noreferrer"
        target="_blank"
      >
        Hooks
      </a>
    </li>
    <li>
      <a
        href="https://reacttraining.com/react-router/web/"
        rel="noopener noreferrer"
        target="_blank"
      >
        React Router
      </a>
    </li>
    <li>
      <a
        href="https://react-bootstrap.github.io/"
        rel="noopener noreferrer"
        target="_blank"
      >
        React Bootstrap
      </a>
    </li>
    <li>
      <a
        href="https://github.com/react-bootstrap/react-router-bootstrap"
        rel="noopener noreferrer"
        target="_blank"
      >
        react-router-bootstrap
      </a>
    </li>
    <li>
      <a
        href="https://sass-lang.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        SASS
      </a>
    </li>
    <li>
      <a
        href="https://reactjs.org/docs/code-splitting.html#route-based-code-splitting"
        rel="noopener noreferrer"
        target="_blank"
      >
        Route based code splitting
      </a>
    </li>
    <li>
      <a
        href="https://redux.js.org/recipes/usage-with-typescript/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Redux using TypeScript
      </a>
    </li>
    <li>
      <a
        href="https://jaredpalmer.com/formik"
        rel="noopener noreferrer"
        target="_blank"
      >
        Forms in React using Formik
      </a>
    </li>
    <li>
      <a
        href="https://github.com/formatjs/react-intl"
        rel="noopener noreferrer"
        target="_blank"
      >
        Support for multiple languages (internationalization) using react-intl
      </a>
    </li>
    <li>
      <a
        href="https://docs.spring.io/spring/docs/current/spring-framework-reference/web-reactive.html"
        rel="noopener noreferrer"
        target="_blank"
      >
        REST services with reactive architecture (non-blocking) with Spring WebFlux and Spring Security
      </a>
    </li>
    <li>
      <a
        href="https://connect2id.com/products/nimbus-jose-jwt"
        rel="noopener noreferrer"
        target="_blank"
      >
        Stateless/sessionless authentication on a server using "Javascript Object Signing and Encryption" (JOSE) and "JSON Web Token" (JWT)"
      </a>
    </li>
    <li>
      <a
        href="https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#use-of-custom-request-headers"
        rel="noopener noreferrer"
        target="_blank"
      >
        Use of custom HTTP headers in stateless/sessionless environments to prevent CSRF attacks
      </a>
    </li>
    <li>
      <a
        href="https://swagger.io/resources/open-api/"
        rel="noopener noreferrer"
        target="_blank"
      >
        OpenAPI 3
      </a>
    </li>
  </ul>
  <p>
    The source code for this project can be found at:
  </p>
  <ul>
    <li>
      Backend: 
      <a
        href="https://github.com/guillermo-varela/multi-player-auth"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://github.com/guillermo-varela/multi-player-auth
      </a>
    </li>
    <li>
      Frontend: 
      <a
        href="https://github.com/guillermo-varela/multi-player-web-reactjs"
        rel="noopener noreferrer"
        target="_blank"
      >
        https://github.com/guillermo-varela/multi-player-web-reactjs
      </a>
    </li>
  </ul>
  <p>
    The multimedia content was fetched using the 
    <a
      href="https://pixabay.com/api/docs"
      rel="noopener noreferrer"
      target="_blank"
    >
      Pixabay API
    </a>
    .
  </p>
</div>
`;
